docs/learn.md

```diff
@@ -721,20 +723,1250 @@ template](https://github.com/melange-re/melange-opam-template).
+type document
+
+external document : document = "document" [@@bs.val]
+```
+
+#### Using functions from other JavaScript modules
+
+`bs.module` is like the `bs.val` attribute, but it accepts a string with the
+name of the module, or the relative path to it.
+
+```ocaml
+external dirname : string -> string = "dirname" [@@bs.module "path"]
+let root = dirname "/User/github"
+```
+
+Generates:
+
+```js
+var Path = require("path");
+var root = Path.dirname("/User/github");
+```
+
+#### Labeled arguments
+
+OCaml has [labeled arguments](https://v2.ocaml.org/manual/lablexamples.html),
+which can also be optional, and work with `external` as well.
+
+Labeled arguments can be useful to provide more information about the arguments
+of a JavaScript function that is called from Melange.
+
+Let’s say we have the following JavaScript function, that we want to call from
+Melange:
+
+```js
+// MyGame.js
+
+function draw(x, y, border) {
+  // let’s assume `border` is optional and defaults to false
+}
+draw(10, 20)
+draw(20, 20, true)
+```
+
+When writing Melange bindings, we can add labeled arguments to make things more
+clear:
+
+```ocaml
+external draw : x:int -> y:int -> ?border:bool -> unit -> unit = "draw"
+  [@@module "MyGame"]
+
+let () = draw ~x:10 ~y:20 ~border:true ()
+let () = draw ~x:10 ~y:20 ()
+```
+
+Generates:
+
+```js
+var MyGame = require("MyGame");
+
+MyGame.draw(10, 20, true);
+MyGame.draw(10, 20, undefined);
+```
+
+The generated JavaScript function is the same, but now the usage in Melange is
+much clearer.
+
+**Note**: in this particular case, a final param of type unit, `()` must be
+added after `border`, since `border` is an optional argument at the last
+position. Not having the last param `unit` would lead to a warning, which is
+explained in detail [in the OCaml
+documentation](https://ocaml.org/docs/labels#warning-this-optional-argument-cannot-be-erased).
+
+Note that you can freely reorder the labels on the Melange side, they will
+appear in the right order on the JavaScript output:
+
+```ocaml
+external draw : x:int -> y:int -> ?border:bool -> unit -> unit = "draw"
+  [@@module "MyGame"]
+let () = draw ~x:10 ~y:20 ()
+let () = draw ~y:20 ~x:10 ()
+```
+
+Generates:
+
+```js
+var MyGame = require("MyGame");
+
+MyGame.draw(10, 20, undefined);
+MyGame.draw(10, 20, undefined);
+```
+
+#### Calling a function as an object property
+
+If we need to call a function that happens to be set as a JavaScript object
+property, Melange provides the attribute `bs.send` so we can define it with just
+one `external`:
+
+```ocaml
+(* Abstract type for the `document` global *)
+type document
+
+external document : document = "document" [@@bs.val]
+external get_by_id : document -> string -> Dom.element = "getElementById"
+  [@@bs.send]
+
+let el = get_by_id document "my-id"
+```
+
+```js
+var el = document.getElementById("my-id");
+```
+
+When using `bs.send`, the first argument will be the object that holds the
```

Comment:
should we also document `bs.send.pipe`, which inserts the argument at the end and is more suitable to be used with `|>`?

Reply:
Added in https://github.com/melange-re/documentation-site/pull/12/commits/80c512d7e8fa2571e645db5ef693d1acbc394585, lmk what you think as it implied quite a bit of changes.

I referred to the blog post I wrote back in the day, maybe we could include other links that explain OCaml pipe `|>`, not sure.